File: versions/forge1122/src/common/java/com/idkidknow/mcreallink/forge1122/mixin/complement/BroadcastingMessage.java
package com.idkidknow.mcreallink.forge1122.mixin.complement;

import net.minecraft.util.text.ITextComponent;

import java.util.function.Consumer;

public class BroadcastingMessage {
    private static final ThreadLocal<Boolean> ignoreMessage = ThreadLocal.withInitial(() -> false);
    public static boolean isIgnoreMessage() {
        return ignoreMessage.get();
    }
    private static Consumer<ITextComponent> callback = component -> {};
    public static Consumer<ITextComponent> getCallback() {
        return callback;
    }
    public static void setCallback(Consumer<ITextComponent> callback) {
        BroadcastingMessage.callback = callback;
    }

    public static void ignoreTemporarily(Runnable action) {
        ignoreMessage.set(true);
        action.run();
        ignoreMessage.set(false);
    }
}

File: src/main/java/com/idkidknow/mcreallink/platform/neoforge121/mixin/complement/BroadcastingMessage.kt
package com.idkidknow.mcreallink.platform.neoforge121.mixin.complement

import net.minecraft.network.chat.Component

/** @see com.idkidknow.mcreallink.platform.neoforge121.mixin.mixin.PlayerListMixin */
object BroadcastingMessage {
    private val _ignoreMessage = ThreadLocal.withInitial { false }
    val ignoreMessage: Boolean get() = _ignoreMessage.get()
    var callback: (Component) -> Unit = {}

    fun ignoreTemporarily(action: () -> Unit) {
        _ignoreMessage.set(true)
        action()
        _ignoreMessage.set(false)
    }
}
